doc-cheat-sheet.rst

```diff
@@ -0,0 +1,600 @@
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+     - Description
+   * - ``.. _target_ID:``
+     -
+     - Adds the target ``target_ID``.
+
+       .. note::
+          When defining the target, you must prefix it with an underscore. Do not use the starting underscore when referencing the target.
+   * - ``:ref:`a_section_target```
+     - :ref:`a_section_target`
+     - References a target that has a title.
+   * - ``:ref:`Link text <a_random_target>```
+     - :ref:`Link text <a_random_target>`
+     - References a target and specifies a title.
+   * - ``:ref:`sphinx-rtd:home```
+     - :ref:`sphinx-rtd:home`
+     - You can also reference targets in other doc sets.
+
+Adhere to the following conventions:
+
+- Never use external links to reference a section in the same doc set or a doc set that is linked with Intersphinx. It would likely cause a broken link in the future.
+- Override the link text only when it is necessary. If you can use the referenced title as link text, do so, because the text will then update automatically if the title changes.
+- Never "override" the link text with the same text that would be generated automatically.
+
+Referencing a page
+^^^^^^^^^^^^^^^^^^
+
+If a documentation page does not have a target, you can still reference it by using the ``:doc:`` role with the file name and path.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - ``:doc:`tutorial/install```
+     - :doc:`tutorial/install`
+   * - ``:doc:`Link text <tutorial/install>```
+     - :doc:`Link text <tutorial/install>`
+   * - ``:doc:`sphinx-rtd:setup-rtd```
+     - :doc:`sphinx-rtd:setup-rtd`
+   * - ``:doc:`Link text <sphinx-rtd:setup-rtd>```
+     - :doc:`Link text <sphinx-rtd:setup-rtd>`
+
+Adhere to the following conventions:
+
+- Only use the ``:doc:`` role when you cannot use the ``:ref:`` role, thus only if there is no target at the top of the file and you cannot add it. When using the ``:doc:`` role, your reference will break when a file is renamed or moved.
+- Override the link text only when it is necessary. If you can use the document title as link text, do so, because the text will then update automatically if the title changes.
+- Never "override" the link text with the same text that would be generated automatically.
+
+Navigation
+==========
+
+Every documentation page must be included as a sub-page to another page in the navigation.
+
+This is achieved with the `toctree <https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-toctree>`_ |x wokeignore:rule=master x| directive in the parent page::
+
+  .. toctree::
+     :hidden:
+
+     sub-page1
+     sub-page2
+
+If a page should not be included in the navigation, you can suppress the resulting build warning by putting ``:orphan:`` at the top of the file.
+Use orphan pages sparingly and only if there is a clear reason for it.
```

Comment:
Is it worth mentioning the exceptions thingy in conf.py as an alternative?

Reply:
That is different though - excluding a file in conf.py will cause it not to be built, while adding `:orphan:` will still build it, but not include it in the navigation.
I'll add a note though that this is another alternative.